cypress/integration/fee-fine/fee-fine-history.spec.js

```diff
@@ -0,0 +1,139 @@
+          feeFineType = { id:  manualCharge.id, name: manualCharge.feeFineType, amount: manualCharge.amount };
+        });
+        PaymentMethods.createViaApi(ownerData.id).then(({ name, id }) => {
+          paymentMethod.name = name;
+          paymentMethod.id = id;
+        });
+        WaiveReasons.createViaApi(getNewWaiveReason()).then(res => {
+          waiveReason.id = res.body.id;
+        });
+
+        RefundReasons.createViaApi(getNewRefundReason()).then(res => {
+          console.log(JSON.stringify(res));
+          refundReason.id = res.body.id;
+        });
+
+        cy.createTempUser(userData.permissions)
+          .then(userProperties => {
+            userData.username = userProperties.username;
+            userData.password = userProperties.password;
+            userData.userId = userProperties.userId;
+            userData.barcode = userProperties.barcode;
+            userData.firstName = userProperties.firstName;
+          })
+          .then(() => {
+            UserEdit.addServicePointViaApi(servicePointId, userData.userId);
+            feeFineAccount = { ownerId: ownerData.id,
+              feeFineId: feeFineType.id,
+              amount: feeFineType.amount,
+              userId: userData.userId,
+              feeFineType: feeFineType.name,
+              createdAt: servicePointId,
+              dateAction: moment.utc().format(),
+              feeFineOwner: ownerData.name,
+              source: 'ADMINISTRATOR, DIKU' };
+            NewFeeFine.createViaApi(feeFineAccount).then((feeFineAccountId) => {
+              feeFineAccount.id = feeFineAccountId;
+              cy.login(userData.username, userData.password);
+            });
+          });
+      });
+  });
+
+
+  after('Delete owner, transfer account, feeFineType, paymentMethod, waiveReason, refundReason, user, feeFine', () => {
+    TransferAccounts.deleteViaApi(transferAccount.id);
+    ManualCharges.deleteViaApi(feeFineType.id);
+    WaiveReasons.deleteViaApi(waiveReason.id);
+    RefundReasons.deleteViaApi(refundReason.id);
+    UsersOwners.deleteViaApi(ownerData.id);
+    Users.deleteViaApi(userData.userId);
+  });
+
+  it('C347919 Check that the user can add "Additional information" on the fee/fine history', { tags: [TestTypes.smoke] }, () => {
+    cy.visit(AppPaths.getFeeFineDetailsPath(userData.userId, feeFineAccount.id)).then(() => {
+      //TODO clarify why this issue sometimes happens
+      if (HTML('Something went wrong').exists()) {
+        cy.visit(AppPaths.getFeeFineDetailsPath(userData.userId, feeFineAccount.id));
+      }
+    });
+    FeeFinesDetails.waitLoading();
+    FeeFinesDetails.openPayModal();
+    PayFeeFaine.checkAmount(feeFineType.amount);
+    PayFeeFaine.setPaymentMethod(paymentMethod);
+    //TODO clarify why the remainAmount sometimes is shown incorrectly
+    PayFeeFaine.setAmount(feeFineType.amount - 1);
+    PayFeeFaine.checkRestOfPay(1);
+    PayFeeFaine.submit();
+    PayFeeFaine.confirm();
```

Comment:
May be better to unite submit and confirm in one fragment?

Reply:
Done.